JustSaying.Sample.Restaurant.OrderingApi/Startup.cs

```diff
@@ -0,0 +1,109 @@
+using Amazon;
+using JustSaying.Sample.Restaurant.Models;
+using JustSaying.Sample.Restaurant.OrderingApi.Handlers;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Swashbuckle.AspNetCore.Swagger;
+
+namespace JustSaying.Sample.Restaurant.OrderingApi
+{
+    public class Startup
+    {
+        private readonly string _awsRegion;
```

Comment:
Access these inline from `IConfiguration` in the builder calls using the overload that receives an `IServiceProvider`, rather than storing here in this class. That would then make it easier to move the builder code out into another class with some extension methods without anchoring it to the `Startup` class.

Reply:
Sorry not too sure what you mean here. 
I've created extension methods for `IConfiguration` so that this startup code can be moved / refactored and the configuration keys are in only one place.

Reply:
I was thinking something like this:

```csharp
services.AddJustSaying((config, serviceProvider) =>
{
    var configuration = serviceProvider.GetRequiredService<IConfiguration>();

    // ...

    config.Messaging(x =>
    {
        x.WithRegion(configuration.GetAWSRegion());
    });

    // ...
}
```

This was initially because I'd misremembered and thought that `IConfiguration` wasn't going to be injectable into `Startup` anymore in 3.0, but it appears that either I was wrong about that, or it's been added back (https://github.com/aspnet/Announcements/issues/353) since an earlier announcement of the changes.

Reply:
ok. shall i leave as is then?